### PR TITLE
Api load and save decklist rework

### PR DESCRIPTION
**Pull request by Benjamin-Hayduchok**

### CODE REVIEW

File: app/components/providers/deckListProvider/DeckListProvider.tsx [hunk 1]
```diff
@@ -1,14 +1,60 @@
...
+  var deckListInit: DeckListType[] = [];
+  for (var cardNum in deckList) {
+    var cardObj = cardsData[parseInt(cardNum) - 1];
+    var cardCount = deckList[cardNum];
+    var parsedCard = parseCardIntoDeckCard(cardObj, parseInt(cardCount));
+    deckListInit.push(parsedCard);
+  }
+  return deckListInit;
+}
+
```
Comment (NIK-VOLO) [inline app/components/providers/deckListProvider/DeckListProvider.tsx]:
```suggestion
function createDeckListObj(deckList: dbDeckListObjType, cardsData: any): DeckListType[] {
  if (!cardsData) {
    return [];
  }

  return Object.entries(deckList).map(([cardNum, cardCount]) => {
    const cardObj = cardsData[parseInt(cardNum) - 1];
    return parseCardIntoDeckCard(cardObj, parseInt(cardCount));
  });
}
```

"The primary advantage of the second version is its improved readability and maintainability, which can often outweigh micro-optimizations in terms of code maintainability and debugging. It also takes advantage of more modern language features, making it more aligned with modern JavaScript practices.

In most practical scenarios, the difference in efficiency between these two versions is likely to be minimal, and you should prioritize code readability and maintainability over minor performance optimizations unless you're working with truly massive data sets."
- Mr. GPT

File: app/components/providers/deckListProvider/DeckListProvider.tsx [hunk 2]
```diff
@@ -1,14 +1,60 @@
...
+const formatDeckObj = (currDeckObj: any) => {
+  var returnObj: { [key: number]: number } = {};
+  for (var card of currDeckObj) {
+    if (card.count > 0) {
+      returnObj[card.cardNum] = card.count;
+    }
+  }
+  return returnObj;
+};
```
Comment (NIK-VOLO) [inline app/components/providers/deckListProvider/DeckListProvider.tsx]:
```suggestion
const formatDeckObj = (currDeckObj: any): { [key: number]: number } => {
  return currDeckObj.reduce((returnObj, { cardNum, count }) => {
    if (count > 0) {
      returnObj[cardNum] = count;
    }
    return returnObj;
  }, {});
};

```

"
We use the `reduce` function to iterate through the `currDeckObj` array and build the `returnObj` object.

The initial value of the reduce function is an empty object {}.

We directly assign properties to the `returnObj` object if the `card.count` is greater than 0.

We use object destructuring to directly extract `cardNum` and `count` from each `card` object within the `currDeckObj` array.

This makes the code more concise and self-explanatory by eliminating the need to access properties like `card.cardNum` and `card.count`.

This is a minor improvement that enhances the readability of your code without sacrificing performance or functionality."

**Review by NIK-VOLO — CHANGES_REQUESTED** [review]
Looks good! Just a few suggested semantic changes :)

File: app/components/providers/deckListProvider/DeckListProvider.tsx (hunk 2 above)
Comment (Benjamin-Hayduchok) [reply app/components/providers/deckListProvider/DeckListProvider.tsx]: Done

File: app/components/providers/deckListProvider/DeckListProvider.tsx (hunk 1 above)
Comment (Benjamin-Hayduchok) [reply app/components/providers/deckListProvider/DeckListProvider.tsx]: Done

**Pull request merged**